ftw/trash/patches.py

```diff
@@ -50,3 +51,20 @@ def searchResults(self, REQUEST=None, **kw):
         kw['trashed'] = [True, False, None]
 
     return self._old_searchResults(REQUEST, **kw)
+
+
+def _getFieldObjects(self, *args, **kwargs):
+    return filter(lambda obj: not ITrashed.providedBy(obj),
+                  self._old__getFieldObjects(*args, **kwargs))
+
+
+def getRawActionAdapter(self, *args, **kwargs):
+    adapter_ids = self._old_getRawActionAdapter(*args, **kwargs)
+    adapter_result = []
+
+    for adapter_id, adapter in map(lambda id_: (id_, getattr(self.aq_explicit, id_, None)),
+                                   adapter_ids):
```

Comment:
I think getting the object in the map could be implemented simpler:
```python
for adapter_id in adapter_ids:
    obj = self.get(adapter_id)
    if not ITrashed.providedBy(adapter):
        adapter_result.append(adapter_id)
```

Or you could use a filter:
```python
adapter_ids = self._old_getRawActionAdapter(*args, **kwargs)
return filter(lambda adapter_id: not ITrashed.providedBy(getattr(self, adapter_id)), adapter_ids)
```

Reply:
I copied this from `Products.PloneFormGen` https://github.com/smcmahon/Products.PloneFormGen/blob/c463a8276f6ff062d50a8e1aae192ed193c66e09/Products/PloneFormGen/content/form.py#L614
```python
getattr(self.aq_explicit, id_, None)
```

Is `get` different to `aq_explicit` if so shouldn't we use the same as they do. Or is `get` better for some reason.

But either way it would be more readable if I'd put the `obj = ..` on a new line.